spec/models/exam_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+RSpec.describe Exam, type: :model do
+  describe "Associations" do
+    it { should belong_to(:quiz) }
+
+    it { should belong_to(:user) }
+
+    it { should have_many(:exam_details).dependent(:destroy) }
+  end
+
+  describe "Exam Nested attributes exam detail" do
+    it { should accept_nested_attributes_for(:exam_details) }
+  end
+
+  describe "scopes" do
+    let(:user) {FactoryBot.create :user}
+    let(:subject) {FactoryBot.create :subject}
+    let(:quiz) {FactoryBot.create :quiz, subject_id: subject.id}
+    let(:exam_1) {FactoryBot.create :exam, quiz_id: quiz.id, created_at: DateTime.now-0.1}
+    let(:exam_2) {FactoryBot.create :exam, quiz_id: quiz.id, created_at: DateTime.now-0.2}
+    let(:exam_3) {FactoryBot.create :exam, created_at: DateTime.now-0.3, user_id: user.id}
```

Comment:
let và let! có khác nhau ko em ơi

Reply:
Theo em đọc được trong doc thì :
- Let:
     `Use let to define a memoized helper method. The value will be cached across multiple calls in the same example but not across examples. Note that let is lazy-evaluated: it is not evaluated until the first time the method it defines is invoked.`
 Có nghĩa là let có giá trị trong từng it block(example) , kèm theo đó là lazy evaluation (  sử dụng 1 biến let nhiều lần trong cùng 1 example)
- Let!
 ` You can use let! to force the method’s invocation before each example. `
 let! được gọi trong một block ẩn, cho nên kết quả đc tính toán và lưu vào cache trước khi thực thi example

link doc:
https://mixandgo.com/learn/ruby/let-vs-let!